source/documentNavigation/sentenceHelper.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: UTF-8 -*-
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited, Rob Meredith
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import re
+from typing import Optional
+
+WESTERN_TERMINATORS_PATTERN: re.Pattern = re.compile("[.?!](?: |\r|\n)+")
+WESTERN_TERMINATORS_SINGLE_PATTERN: re.Pattern = re.compile("[.?!]")
+FULL_WIDTH_TERMINATORS_PATTERN: re.Pattern = re.compile("[。！？](?:\r|\n)*")
```

Comment:
@rob-aph Are these regex differences expected?

Reply:
I think that there is a simpler way to write the regexp string for western sentence ending, taking into account both cases as follow:
`"[.?!](?: |\r|\n|$)+"`

This would allow to avoid duplicating some information, e.g. the 3 punctuation symbols denoting the end of a sentence.

Reply:
@CyrilleB79 Thanks for the optimization. I didn't realize that $ can be ored in with other characters and thought it needed to stand alone.